pkg/core/common.go

```diff
@@ -14,20 +14,92 @@
 package core
 
 import (
+	"encoding/json"
+	"strings"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type ObjectBase struct {
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+	Kind      string `json:"kind"`
+	UID       string `json:"uid"`
+	Created   string `json:"created_at"`
+}
+
 // KubeObjectDesc defines a simple kube object description which uses in apiserver.
 type KubeObjectDesc struct {
-	metav1.TypeMeta `json:",inline"`
-	Meta            KubeObjectMeta `json:"metadata"`
-	Spec            interface{}    `json:"spec"`
+	metav1.TypeMeta
+	Meta KubeObjectMeta `json:"metadata"`
+	Spec interface{}    `json:"spec"`
 }
 
 // KubeObjectMetadata extracts the required fields from metav1.ObjectMeta.
 type KubeObjectMeta struct {
-	Name        string            `json:"name"`
 	Namespace   string            `json:"namespace"`
+	Name        string            `json:"name"`
 	Labels      map[string]string `json:"labels,omitempty"`
 	Annotations map[string]string `json:"annotations,omitempty"`
 }
+
+type Filter struct {
+	ObjectID  string `json:"object_id"`
+	Start     string
```

Comment:
how about adding ``json:"-"`` for `Start`, `End` and `Limit`?

Reply:
This will always omit the fields, but we need them.